Documentation/README.md

```diff
@@ -57,3 +57,130 @@ tar -xf <tarball-destination> -C <path-to-extraction-root>
 cd <path-to-extraction-root>
 ./build.sh
 ```
+## Building a custom .NET Runtime
+
+You can also build a .NET Core Runtime with custom changes in order to test locally the produced runtime with the affected changes.
+
+The .NET Core Runtime consists of 4 repos, core-setup at the top of the stack (which produces the runtime), standard, corefx (BCL repo), coreclr (Runtime repo). 
+
+This means that in order to get a runtime you'll need to build the source-build repo with the `/p:RootRepo=core-setup` parameter, this will automatically build coreclr, standard and corefx. Source-Build is composed by git submodules, so the first thing you need to run before doing anything is:
+
+```console
+git submodule --init --recursive
+```
+
+This will initialize the submodules and clone the repos under the src folder if they haven't been initialized. Note that this will point the submodule to the branch and commit that the submodule is pointing to. However you can go to each repo indivually, add custom remotes and checkout certain branches containing your changes.
+
+Let's say that you want to produce a runtime that has a new API that it's implementation lives in coreclr and you want to create a .NET Core console app that uses it to test it without waiting for a new SDK or Runtime to be produced by official builds. Imagine we will add an API that just throws in Hashtable. 
+
+Hashtable implementation lives in coreclr and it's APIs are exposed in corefx in the System.Runtime.Extensions.cs contract. So in my local coreclr repo I will just add Hashtable.cs the following API:
+
+```cs
+public static void PNSE()
+{
+    throw new PlatformNotSupportedException();
+}
+```
+
+Then in my local corefx repo let's expose it in System.Runtime.Extensions.cs in the Hashtable class:
+
+```cs
+public static void PNSE() { }
+```
+
+Of course we need to commit this changes in our current repos. Once we've commited this locally, let's move to the source-build repo.
+
+Once in the source-build in our commmand prompt or terminal, lets navigate to `src/coreclr`. Once there, we will need to add a remote to the local coreclr repo, in example:
+```console
+git remote add local C:\repos\coreclr
+```
+
+Then, let's fetch the changes from our repo, by running:
+```console
+git fetch local
+```
+
+Once we've fetch the changes, we just need to checkout the branch where we made the changes:
+```console
+git checkout branchWithChanges
+```
+
+Just repeat this steps under `src/corefx` as well, to pull the corefx changes. Once we've done that, let's navigate to the `source-build` root and run:
+
+```console
+build.{cmd|sh} /p:RootRepo=core-setup
+```
+
+Note that since you've changed the coreclr and corefx commits, the incremental build will find the current commit changed so if you don't want to see messages like:
+```console
+WARNING: submodule src/corefx, currently at 831264e53e5b9333850baa659af8a2857a9cb9b7, has diverged from checked-in
+version 5b7674e4ae5cc782e99f50b2919dfdeb29106a46
```

Comment:
We should file an issue about this because we don't really want folks to be required to stage local changes to build.

Reply:
Just to make sure you see: `export SOURCE_BUILD_SKIP_SUBMODULE_CHECK=1`/`$env:SOURCE_BUILD_SKIP_SUBMODULE_CHECK = 1` will skip the check entirely, and the user can press `n` to continue. (Can the message be improved?)

Reply:
Would be worth to add a msbuild property to disable this through an msbuild parameter? Whenever we improve this.

Reply:
The check runs in `build.sh/ps1` before running msbuild (and we pass args through the script without parsing any of them), so we can only do that if we move the whole thing to somewhere later.

On linux you can say `SOURCE_BUILD_SKIP_SUBMODULE_CHECK=1 ./build.sh` to avoid exporting it, for what it's worth.

Reply:
I had an idea that might help: https://github.com/dotnet/source-build/issues/613. Keep the check, but change the default to notify and let the user configure it to prompt or disable if they want.

Reply:
I think it should stay in until someone changes the behavior, since people will *definitely* see it while following these instructions and wonder what to do. If it were a huge section maybe it would make sense to remove it, but it's only a few short paragraphs.

Reply:
I agree with calling out the environment variable here, we can change the message as well if there's better phrasing.  This runs before MSBuild gets involved so it'll have to stay an environment variable unless we want to start calling out to the check-submodules script from an MSBuild project.

Reply:
Ack, sorry, GitHub didn't update before I commented.